app/models/topic.rb

```diff
@@ -131,4 +133,12 @@ def topic_autocomplete_method
   def mk_permission
     Perm.short(permission)
   end
+
+  protected
+    def create_metamap?
+      if link == '' and metacode.name == 'Metamap'
+        @map = Map.create({ name: name, permission: permission, desc: '', arranged: true, user_id: user_id })
+        self.link = Rails.application.routes.url_helpers.map_url(:host => ENV['MAILER_DEFAULT_URL'], :id => @map.id)
```

Comment:
This still niggles me. Are we able to make the link relative instead of absolute? I guess not, if we want the topic to be exportable. I'll try and find a way to do this maybe, but don't let it block merging